docs/i18n/10-bundles.md

```diff
@@ -0,0 +1,131 @@
+* Provides a comprehensive set of message keys and their content (represented in the default language) that are used as a fallback if other languages within the bundle do not provide overrides for a given key
+* Lists the bundle's other supported languages, as well as the mechanism to load the set of messages from each supported language's module
+
+## TypeScript structure
+
+Every language within a bundle is a TypeScript module, and is required to export a default object containing a `messages` property. This property should be a map of message keys to their translated values within the particular language.
+
+For example, an English language module within a bundle:
+
+```ts
+export default {
+	messages: {
+		hello: 'Hello',
+		goodbye: 'Goodbye'
+	}
+};
+```
+
+### Default language module
+
+The language module designated as the bundle's default should also provide a `locales` property on its exported object. This property is a map of locale identifiers to functions that can load the message set for each language/locale supported by the bundle.
+
+For example, a bundle supporting French, Arabic and Japanese (with English designated as default):
+
+```ts
+export default {
+	locales: {
+		fr: () => import('./fr/main'),
+		ar: () => import('./ar/main'),
+		ja: () => import('./ja/main')
+	},
+	messages: {
+		hello: 'Hello',
+		goodbye: 'Goodbye'
+	}
+};
+```
+
+The `messages` object within the bundle's default language module contains a comprehensive set of all message keys and their descriptions for a particular section of an application. When the application locale is set to the default value, the bundle's default language module acts as a regular message lookup via its own `messages` property. When a non-default locale is in use, the default's `messages` are used as fallbacks for any keys not included in the bundle's additional language modules.
+
+Each bundle's default language module will also have a unique `id` property assigned to it. The bundle ID is used internally to manage caching and handle interoperability with other parts of the i18n infrastructure. While it is possible for bundles to specify their own `id` property alongside `locales` and `messages` within the bundle's default language module, doing so is neither necessary nor recommended.
+
+## Importing and using bundles
+
+The default language module for a bundle is `import`ed like any other TypeScript module into each widget that requires use of the set of messages contained within the bundle.
+
+For example, given a default bundle:
+>nls/MyI18nWidget.en.ts
+```ts
+export default {
+	messages: {
+		hello: 'Hello',
+		welcome: 'Welcome to your application'
+	}
+};
+```
+
+This can be imported and referenced within a widget such as:
+>widgets/MyI18nWidget.ts
+```ts
+import { WidgetBase } from '@dojo/framework/widget-core/WidgetBase';
+import { v } from '@dojo/framework/widget-core/d';
+import I18nMixin from "@dojo/framework/widget-core/mixins/I18n";
```

Comment:
There is a mixture of `"` and `'` in some of the examples

Reply:
Fixed